presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -122,19 +123,19 @@ public Expression rewrite(FieldOrExpression fieldOrExpression)
         }
     }
 
-    public void put(Expression expression, Symbol symbol)
+    public void put(Expression expression, Symbol symbol, Collection<Expression> columnReferences)
```

Comment:
@martint , we will always put the mapping into expressionMappings. The columnReferences is used to determine if we need to put resolvedName index for the expression the into fieldSymbols.